requires.go

```diff
@@ -7,20 +7,15 @@ import (
 	"github.com/go-task/task/v3/taskfile/ast"
 )
 
-func (e *Executor) areTaskRequiredVarsSet(t *ast.Task, call *ast.Call) error {
+func (e *Executor) areTaskRequiredVarsSet(t *ast.Task) error {
 	if t.Requires == nil || len(t.Requires.Vars) == 0 {
 		return nil
 	}
 
-	vars, err := e.Compiler.GetVariables(t, call)
-	if err != nil {
-		return err
-	}
-
```

Comment:
Let me know if I'm missing anything, but I think we can just skip evaluation completely for performance. For `requires`, it's enough to know if it is set or not.

Reply:
Can't think of any reason. This is a nice optimisation 👍